lang/elaborator/src/typechecker/decls/mod.rs

```diff
@@ -8,17 +8,34 @@ mod global_let;
 
 use ast::*;
 
-use super::{ctx::Ctx, lookup_table::build_lookup_table, TypeError};
+use super::{
+    ctx::Ctx,
+    lookup_table::{build_lookup_table, LookupTable},
+    TypeError,
+};
 
 pub fn check(prg: Rc<Module>) -> Result<Module, TypeError> {
-    let lookup_table = build_lookup_table(&prg);
-    let mut ctx = Ctx::new(prg.meta_vars.clone(), lookup_table, prg.clone());
+    let mut lookup_table = Default::default();
```

Comment:
We can remove the `check` function, since it isn't used anywhere. We always use `check_with_lookup_table`.